### PR TITLE
Remove ts-interface checking, add stats HTML, modern target

### PR DESCRIPTION
**Pull request by gzuidhof**
A first step towards slimming down Zarr

### CODE REVIEW

File: rollup.config.ts [hunk 1]
```diff
@@ -2,20 +2,32 @@ import resolve from '@rollup/plugin-node-resolve';
...
+      entryFileNames: '[name].min.mjs',
+      sourcemap: true,
+      plugins: [terser()]
+    },
+  ],
     watch: {
       include: 'src/**',
     },
-    plugins: [typescript({ useTsconfigDeclarationDir: true }), commonjs(), resolve()],
+    plugins: [...commonPlugins(), visualizer({filename: "stats.html"}), visualizer({filename: "stats.min.html", sourcemap: true})],
```
Comment (manzt) [inline rollup.config.ts:30]: Can `plugins === commonPlugins()` here and `output[0].plugins === [ visualizer({filename: "stats.html"}) ]` and `output[1].plugins === [ terser(), visualizer({filename: "stats.min.html", sourcemap: true }) ]`

File: rollup.config.ts [hunk 2]
```diff
@@ -31,13 +43,7 @@ export default [
...
     ],
     plugins: [
-      typescript({
-        useTsconfigDeclarationDir: true,
-        // https://github.com/ezolenko/rollup-plugin-typescript2/issues/105
-        objectHashIgnoreUnknownHack: true,
-      }),
-      commonjs(),
-      resolve(),
+      ...commonPlugins()
```
Comment (manzt) [inline rollup.config.ts:46]: nit: no need to spread, just `plugins === commonPlugins()`, unless you think leaving room for future node-specific plugins is useful.

**Review by manzt — APPROVED** [review]
Nice, thanks for taking the first steps in this direction. Love the updated ES target; I think it's perfect for something like this package. 

Odds are if you're using Zarr.js, it will be in a modern browser, and if not, you can transpile with your own build tools.

File: rollup.config.ts (hunk 1 above)
Comment (gzuidhof) [reply rollup.config.ts:30]: The visualizer plugin didn't want to be an output plugin so I had to use this workaround

Comment (manzt) [reply rollup.config.ts:30]: Ok, not a big deal :)

**Comment by manzt** [comment]
Just wanted to follow up on this. Any thing I can do to land this PR and create a new release? This is a great first step, thank you!

**Comment by gzuidhof** [comment]
Sorry for the late reply - I'm happy to merge it as is and then we can mint a new release :).

**Pull request merged**